File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/li_std_wstring_runme.cs
using System;
using li_std_wstringNamespace;

public class runme
{
    static void Main() 
    {
      char y='h';

      if (li_std_wstring.test_wcvalue(y) != y)
        throw new Exception("bad string mapping:" + li_std_wstring.test_wcvalue(y));

      if (li_std_wstring.test_wcvalue_w() != 'W')
        throw new Exception("bad string mapping:" + li_std_wstring.test_wcvalue_w());

      string x="hello";

      if (li_std_wstring.test_ccvalue(x) != x)
        throw new Exception("bad string mapping");

      if (li_std_wstring.test_cvalue(x) != x)
        throw new Exception("bad string mapping");


      if (li_std_wstring.test_value(x) != x)
        throw new Exception("bad string mapping: " + x + li_std_wstring.test_value(x));

      if (li_std_wstring.test_const_reference(x) != x)
        throw new Exception("bad string mapping");


      string s = "he";
      s = s + "llo";

      if (s != x)
        throw new Exception("bad string mapping: " + s + x);

      if (li_std_wstring.test_value(s) != x)
        throw new Exception("bad string mapping");

      if (li_std_wstring.test_const_reference(s) != x)
        throw new Exception("bad string mapping");

      string a = s;

      if (li_std_wstring.test_value(a) != x)
        throw new Exception("bad string mapping");

      if (li_std_wstring.test_const_reference(a) != x)
        throw new Exception("bad string mapping");

      string b = " world";

      if (a + b != "hello world")
        throw new Exception("bad string mapping");

      if (a + " world" != "hello world")
        throw new Exception("bad string mapping");

      if ("hello" + b != "hello world")
        throw new Exception("bad string mapping");

      s = "hello world";

      B myB = new B("hi");

      myB.name = "hello";
      if (myB.name != "hello")
        throw new Exception("bad string mapping");

      myB.a = "hello";
      if (myB.a != "hello")
        throw new Exception("bad string mapping");
    }
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/li_std_wstring_runme.py
import li_std_wstring

x = u"h"

if li_std_wstring.test_wcvalue(x) != x:
    print li_std_wstring.test_wcvalue(x)
    raise RuntimeError("bad string mapping")

x = u"hello"
if li_std_wstring.test_ccvalue(x) != x:
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_cvalue(x) != x:
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_wchar_overload(x) != x:
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_wchar_overload("not unicode") != "not unicode":
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_value(x) != x:
    print x, li_std_wstring.test_value(x)
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_const_reference(x) != x:
    raise RuntimeError("bad string mapping")


s = li_std_wstring.wstring(u"he")
s = s + u"llo"

if s != x:
    print s, x
    raise RuntimeError("bad string mapping")

if s[1:4] != x[1:4]:
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_value(s) != x:
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_const_reference(s) != x:
    raise RuntimeError("bad string mapping")

a = li_std_wstring.A(s)

if li_std_wstring.test_value(a) != x:
    raise RuntimeError("bad string mapping")

if li_std_wstring.test_const_reference(a) != x:
    raise RuntimeError("bad string mapping")

b = li_std_wstring.wstring(" world")

if a + b != "hello world":
    raise RuntimeError("bad string mapping")

if a + " world" != "hello world":
    raise RuntimeError("bad string mapping")

# This is expected to fail if -builtin is used
# Reverse operators not supported in builtin types
if not li_std_wstring.is_python_builtin():
    if "hello" + b != "hello world":
        raise RuntimeError("bad string mapping")

    c = "hello" + b
    if c.find_last_of("l") != 9:
        raise RuntimeError("bad string mapping")

s = "hello world"

b = li_std_wstring.B("hi")

b.name = li_std_wstring.wstring(u"hello")
if b.name != "hello":
    raise RuntimeError("bad string mapping")


b.a = li_std_wstring.A("hello")
if b.a != u"hello":
    raise RuntimeError("bad string mapping")
